Parameter to script

fabiobott opened the pull request:
Is it possible to send parameters to open script, stop script and close script?
I have 12 covers and I wouldn't do 36 scripts.
Thanks

nagyrobi committed:
Update README.md

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update README.md

nagyrobi committed:
Rename custom_components/cover_time_based/cover.py to custom_components/cover_rf_time_based/cover.py

nagyrobi committed:
Rename custom_components/cover_time_based/manifest.json to custom_components/cover_rf_time_based/manifest.json

nagyrobi committed:
Rename custom_components/cover_time_based/services.yaml to custom_components/cover_rf_time_based/services.yaml

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
auto_stop_if_necessary commented out

when covers are operated by triggers like RF, if an RF command was missed by the cover for some reason and because there's no feedback from the cover we can't know what's the real state, this blocks us from re-sending the same command.

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Create hacs.json

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Delete .gitignore

nagyrobi committed:
Delete .gitattributes

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Create manifest.json

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Delete manifest.json

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Renaming CoverDevice class call to CoverEntity

CoverDevice class name was deprecated in last HA.

nagyrobi committed:
Add support for send_stop_at_ends option

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi commented:
Not possible, sorry.
I also have many covers (11). Making scripts for each is super-easy. Just create them for one of the covers, test them thoroughly, and after that just Ctrl+C and Ctrl+V for each and just replace the names accordingly.
If you [split up your configuration](https://www.home-assistant.io/docs/configuration/splitting_configuration/) you can have the scripts in separate files/folders making things easy to look after.

As I've noted several times, the requirement to create the scripts is a way to stimulate people to adapt the component to their custom requirements straight in Home Assistant, without the need to learn programming in Py. One can create a Home Assistant script for anything. You can have any customly developed thing over there to move your curtains which easily embed in your script and then hook it into the time based cover component. You can even embed automations into scripts etc. this gives best flexibility.

There are literally dozens of cover-moving systems out there which weren’t designed originally to be integrated into a smart home system, it’s nearly impossible to create a custom component for each type. Now this is a more universal solution for such cases.

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

VDRainer committed:
Fix example config and mqtt relay stop script

The stop script should first cut the power relay.
If it first stops the open/close relay it's possible that the cover moves a short time (possible mqtt or script delays) back.

nagyrobi committed:
Merge pull request #4 from VDRainer/patch-2

Fix example config and mqtt relay stop script

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update README.md

Alfiegerner committed:
Adding service to expose service that allows setting of position.

This is useful for when sensors can detect the real cover position, which may have been altered outside of HA Control.

Example use case:
  Door sensor gets activated when cover is fully closed, this triggers the service call setting the cover position to 0.

The service takes:

  -- entity_id
  -- position
  -- confident - a boolean which indicates that we know the given position confidentally, which alters assumed_state to False.

The internally maintained state of confidence we have in our poistion is reset to false whenever the component triggers an action.

Alfiegerner committed:
Fix typo colon

Alfiegerner committed:
Use entity_ids in schema rather entity_id to make configurable in UI automations more easily.

Alfiegerner committed:
Fix bug with assumed state and update README.

Alfiegerner committed:
Tidy up README

Alfiegerner committed:
Fix incorrectly cut code.

nagyrobi committed:
Merge pull request #11 from Alfiegerner/master

Adding service that allows setting of position with an position confidence indicator.

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

Alfiegerner committed:
Add position_type parameter to allow setting of either current or target position.

Alfiegerner committed:
Text typo

Alfiegerner committed:
Merge remote-tracking branch 'upstream/master'

Merge conflict README

Alfiegerner committed:
More README changes.

Alfiegerner committed:
More typos

Alfiegerner committed:
Last one ...

Alfiegerner committed:
Add restore functionality for assumed state.  If cover is known to be closd and state is confirmed correct don't lose that on restart.

Slightly odd implementation - using Booleans means that property is not persisted in old_state if False.  To avoid confusion persist a new string property.

nagyrobi committed:
Merge pull request #12 from Alfiegerner/master

Adding position_target parameter to set_known_position service

nagyrobi committed:
Update services.yaml

nagyrobi committed:
Update cover.py

nagyrobi committed:
Create info.md

nagyrobi committed:
Update info.md

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Merge pull request #13 from Alfiegerner/master

Add restore functionality for assumed state.  If cover is known to be…

Alfiegerner committed:
Add new service set_known_action to update HA cover with known actions without triggering the action itself.

Alfiegerner committed:
Merge remote-tracking branch 'upstream/master'

update cover.py with new.

Alfiegerner committed:
ReadMe styles.

Alfiegerner committed:
spacing

Alfiegerner committed:
remove now unused state fir action.

Alfiegerner committed:
Odd instance where attribute is rendered as bool or str - haven't worked it out so catching both.

nagyrobi committed:
Merge pull request #14 from Alfiegerner/master

Add known action service

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

Alfiegerner committed:
Fix condfidence on restore bug.

nagyrobi committed:
Merge pull request #15 from Alfiegerner/master

  Fix condfidence on restore bug.

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update manifest.json

Fix wrong knx version.

barmazu committed:
integrate with latest xknx version - reverted cover directions

nagyrobi committed:
Merge pull request #21 from barmazu/knx_0_17_1_integration

integrate with latest xknx version - reverted cover directions

nagyrobi committed:
Update README.md

Add infos about code repetition options.

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Revert "integrate with latest xknx version - reverted cover directions"

nagyrobi committed:
Merge pull request #23 from nagyrobi/revert-21-knx_0_17_1_integration

Revert "integrate with latest xknx version - reverted cover directions"

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update info.md

nagyrobi committed:
add unique_id property to the covers

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update README.md

regevbr committed:
add support for cover entity

regevbr committed:
add support for cover entity

regevbr committed:
add support for cover entity

regevbr committed:
add support for cover entity

regevbr committed:
add support for cover entity

nagyrobi committed:
Merge pull request #37 from regevbr/cover

add support for cover entity

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update info.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update info.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update cover.py

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update cover.py

- add new `device_class` configuration option

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update manifest.json

wrt54g committed:
Update HACS URL

nagyrobi committed:
Merge pull request #42 from wrt54g/hacs-url

Update HACS URL

nagyrobi committed:
Availability template added

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Remove old xknx dependency

Still relying on the old TravelCalculator method from 0.9.4 - which works well here.
To avoid xknx version conflicts, as per @farmio suggestion dependency is removed by copying these functions to the component.

nagyrobi committed:
Update README.md

nagyrobi committed:
Update hacs.json

nagyrobi committed:
Update info.md

nagyrobi committed:
Update README.md

nagyrobi committed:
Update manifest.json

nagyrobi committed:
Update README.md

nagyrobi committed:
Update README.md
